src/rulesets/asyncapi/index.json

```diff
@@ -0,0 +1,435 @@
+{
+  "formats": [
+    "asyncapi2"
+  ],
+  "rules": {
+    "asyncapi2-channel-no-empty-parameter": {
+      "description": "Channel path should not have empty parameter substitution pattern.",
+      "recommended": true,
+      "type": "style",
+      "given": "$.channels.*~",
+      "then": {
+        "function": "pattern",
+        "functionOptions": {
+          "notMatch": "{}"
+        }
+      }
+    },
+    "asyncapi2-channel-no-query-nor-fragment": {
+      "description": "Channel path should not include a query (`?`) or a fragment (`#`) delimiter.",
+      "recommended": true,
+      "type": "style",
+      "given": "$.channels.*~",
+      "then": {
+        "function": "pattern",
+        "functionOptions": {
+          "notMatch": "[\\?#]"
+        }
+      }
+    },
+    "asyncapi2-channel-no-trailing-slash": {
+      "description": "Channel path should not end with a slash.",
+      "recommended": true,
+      "type": "style",
+      "given": "$.channels.*~",
+      "then": {
+        "function": "pattern",
+        "functionOptions": {
+          "notMatch": ".+\\/$"
+        }
+      }
+    },
+    "asyncapi2-headers-schema-type-object": {
+      "description": "Headers schema type should be `object`.",
```

Comment:
this is a specification requirement (https://www.asyncapi.com/docs/specifications/2.0.0/#a-name-messageobject-a-message-object), the parser will find out about this issue. Are those also parts that spectral ruleset should check?

Reply:
@derberg I may need some help on this topic:

Given the following document

```
  {
      "asyncapi": "2.0.0",
      "info": {
          "title": "Valid AsyncApi document",
          "version": "1.0"
      },
      "channels": {},
      "components": {
          "messages": {
              "userSignUp": {
                  "summary": "Action to sign a user up.",
                  "description": "Multiline description of what this action does.\nHere you have another line.\n",
                  "tags": [{
                          "name": "user"
                      },
                      {
                          "name": "signup"
                      }
                  ],
                  "headers": {
                      "type": "integer",
                      "enum": [1, 2]
                  },
                  "payload": {
                      "type": "integer"
                  }
              }
          }
      }
  }
```

The asyncapi2 schema doesn't seem to catch the wrong `headers` type.

Help?

Reply:
@nulltoken I would say it is a bug in our parser that doesn't validate header, spec is clear here, should be `object`

Reply:
I reported a bug for us https://github.com/asyncapi/parser-js/issues/68

Reply:
@derberg Awesome! I subscribed to the issue. When it's closed, I'll drop this rule.